models.py

```diff
@@ -69,6 +72,12 @@ def __init__(self, **kwargs):
         self.merchant_code = generate_key(4)
         super().__init__(**kwargs)
 
+    def generate_defaults(self):
```

Comment:
change this function to 'on_admin_created', remove merchant_code initialization because it is already done in the constructor

Reply:
im pretty sure the merchant_code doesn't get generated when going via create form. Will test.

Reply:
so it looks like when adding the user via CLI the difference is this:
    if len(sys.argv) > 1:
        if sys.argv[1] == "add_user":
            add_user(sys.argv[2], sys.argv[3])
        if sys.argv[1] == "add_role":
            add_role(sys.argv[2], sys.argv[3])
        if sys.argv[1] == "add_merchant_codes":
            add_merchant_codes()

So the merchant_codes being added when the you're adding the user or role, as long as there is more than 1 argument.

Reply:
ok just rename the function then